src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -168,7 +168,8 @@ public ObservableList<Internship> getFilteredInternshipList() {
     @Override
     public void updateFilteredInternshipList(Predicate<Internship> predicate) {
         requireNonNull(predicate);
-        filteredInternships.setPredicate(predicate);
+        FilteredList<Internship> internshipList = (FilteredList<Internship>) filteredInternships.getSource();
```

Comment:
```suggestion
        // Since we are just swapping between 2 observable lists and they are wrappers around
        // the source list, it is safe to swap between SortedList and FilteredList.
        @SuppressWarnings("unchecked")
        FilteredList<Internship> internshipList = (FilteredList<Internship>) filteredInternships.getSource();
```